scripts/sort-makefile-am-sources.pl

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/perl
+#
+## Copyright (C) 1996-2020 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+use strict;
+use warnings;
+
+while (<>) {
+    if (m!^#!) {
+        print;
+        next;
+    }
+    if (/^(\S+_SOURCES)\s*=\s*\\$/) {
+        print "$1 = \\\n";
+    } else {
+        print;
+        next;
+    }
+    # accumulate files and prep for sorting
+    my %files;
+    while (<>) {
+        my $prefix='';
+        my $filename='';
+
+        chomp;
+        m!\s*(tests/stub_|tests/test)?(\S+)(\s+\\\s*)?$! || die "no parse";
+        $prefix=$1 if (defined $1);
+        $filename=$2 if (defined $2);
+        
+        $files{"$filename.$prefix"}="$prefix$filename";
+        if (! /\\$/ ) {  # last line in the list
+            &print_files(\%files);
+            last;
+        }
+    }
+}
+
+# arg is hash ref, print values in order of key
+sub print_files
+{
+    my %files=%{$_[0]};
+    my @q=();
+    foreach my $k (sort {lc $a cmp lc $b} keys %files) {
```

Comment:
This is too clever to avoid doubts IMO.

```suggestion
    foreach my $k (sort {(lc $a) cmp (lc $b)} keys %files) {
```

Reply:
Also, using lc for case-insensitive comparison is creating visually confusing sort orders. Some files put "Con*" then "con*" others "con*" then "Con*" depending on characters far down the paths.

IMO it is better to be case sensitive and let devs easily predict what position new additions should be put to avoid followup enforcement commits.

Reply:
removing lowercasing as suggested by @yadij